src/odemis/util/test/img_test.py

```diff
@@ -1181,8 +1183,873 @@ def test_rotate_img_metadata(self):
+        # keep the z-slice at default
+        data = model.DataArray(self.arr1[0, 0, :], metadata=self.md)
+        output_DA = numpy.asarray(img.getYXFromZYX(data))
+
+        # check if the XY slice output is part of input DA
+        self.assertTrue(output_DA in data)
+        # check if the slice has the same size (x,y) as input
+        self.assertEqual(output_DA.shape[1], data.shape[0])
+        # check if all elements are of the set color_num
+        self.assertTrue(numpy.all(output_DA == self.color_num))
+
+    def test_get_yx_from_zyx_4D(self):
+        """
+        Verify that using a 4D DA picture will fail.
+        see if a ValueError exception is raised
+        """
+        # use a 4D array to force a ValueError when slicing array shape
+        data = model.DataArray(self.arr1.reshape(20, 399, 399, 10), metadata=self.md)
+
+        with self.assertRaises(ValueError):
+            img.getYXFromZYX(data)
+
+
+class TestEnsure2DImage(unittest.TestCase):
+    # input data array
+    # output 2D data array
+    # test for different dimension values
+    # test forcing ValueError
+
+    def setUp(self):
+        y = 4000
+        x = 4000
+        # size of picture in CTZYX
+        size = (1, 1, 1, y, x)
+        arr = numpy.ones(size, dtype='uint8')
+        self.data = model.DataArray(arr)
+
+    def test_ensure_2d_image_2d(self):
+        """
+        Verify that using a 5D CTZYX DA will return the
+        right 2D DA.
+        normal use of function
+        """
+        out_arr = numpy.asarray(img.ensure2DImage(self.data))
+
+        # check if the returned array shape is 2D
+        self.assertTrue(out_arr.ndim == 2)
+        # check if the YX values match
+        self.assertEqual(out_arr.shape, self.data.shape[3:])
+
+    def test_ensure_2d_image_5d(self):
+        """
+        Verify that using a 5D DA will fail.
+        see if a ValueError exception is raised
+        """
+        self.data = numpy.asarray(self.data).reshape(1, 2, 50, 400, 400)
+
+        with self.assertRaises(ValueError):
+            img.ensure2DImage(self.data)
+
+
+class TestSubtract(unittest.TestCase):
+    # input data array
+    # output 2D data array
+    # test for dtype unsigned int, float, combi
+
+    def setUp(self):
+        self.pic1 = numpy.zeros((600, 600))
+        self.pic2 = numpy.zeros((600, 600))
+
+    def test_subtract_uint(self):
+        """
+        Verify that using an uint8 as dtype will return the right DA.
+        normal use of function
+        """
+        pic1 = self.pic1.astype('uint8')
+        pic2 = self.pic2.astype('uint8')
+        # add white squares in the black pictures
+        pic1[200:500, 200:500] = 255
+        pic2[300:400, 300:400] = 255
+
+        # subtract the pictures
+        out_arr = img.Subtract(pic1, pic2)
+
+        # check if the subtraction worked by finding the smaller
+        # square of pic2 in the output picture
+        self.assertEqual(out_arr.shape, pic1.shape, pic2.shape)
```

Comment:
You cannot check that all of them are the same shape like that. The 3rd argument is the message to display in case there is an error.
Anyway, just check the shape of out_arr and pic1 is enough.